Update ece-architecture.md - Adding a container <-> ECE host role mapping table

rseldner opened the pull request:
**DRAFT** Proposing we document the actual service container names with a brief description in a `container`<->`ece host role` mapping table.  `ece-architecture.md` seems like an appropriate place for this.  Kept it brief to limit redundancies from the sections above. 

Preview: https://docs-v3-preview.elastic.dev/elastic/docs-content/pull/1369/deploy-manage/deploy/cloud-enterprise/ece-architecture

rseldner committed:
Update ece-architecture.md

Proposing we document the actual service container names in a container-to-role mapping table

jakommo commented on deploy-manage/deploy/cloud-enterprise/ece-architecture.md:
```diff
@@ -84,3 +84,26 @@ Services are deployed as Docker containers, which simplifies the operational eff
+| `frc-directors-director`                            | Director      | Manages ZooKeeper and internal certificates; maintains Stunnel and quorum. |
+| `frc-proxies-proxyv2`                               | Proxy         | Routes user traffic; uses ZooKeeper to map deployment IDs to cluster nodes. |
+| `frc-proxies-route-server`                          | Proxy         | Manages internal routing and port mapping for cluster access. |
+| `frc-runners-runner`                                | All roles     | Deploys containers locally per host role definitions. |
```
```suggestion
| `frc-runners-runner`                                | All roles     | Runs on every ECE host and provides a supervisor service to deploy and manage containers according to their defined roles, ensuring they are online and healthy. |
```

jakommo commented on deploy-manage/deploy/cloud-enterprise/ece-architecture.md:
```diff
@@ -84,3 +84,26 @@ Services are deployed as Docker containers, which simplifies the operational eff
+| `frc-allocators-allocator`                          | Allocator     | Manages container lifecycle for Elasticsearch and Kibana; reports host capacity to ZooKeeper. |
+| `frc-beats-runners-beats-runner`                    | All roles     | Collects metrics from local containers for monitoring and health checks. |
+| `frc-blueprints-blueprint`                          | Director      | Coordinates container startup by providing configuration data to runners based on their role and token. |
+| `frc-client-forwarders-client-forwarder`            | All roles     | Manages communication between nodes and ZooKeeper. |
```
```suggestion
| `frc-client-forwarders-client-forwarder`            | All roles     | Manages communication between hosts and ZooKeeper. |
```
I think `node` is already used for too many different things and thus suggesting host. Another option could be Runner, but host is more generic and doesn't require ECE knowledge

rseldner replied:
Agreed with the term `host` and what I would usually use when speaking ECE.

jakommo reviewed (approved):
LGTM, thanks for creating this @rseldner !

In general, I'd appreciate if someone from the docs team could go over this re terms used. I.e. cluster is used a few times, where I think deployment could be a better term to stay within ECE terminology ...

eedugon commented on deploy-manage/deploy/cloud-enterprise/ece-architecture.md:
```diff
@@ -84,3 +84,26 @@ Services are deployed as Docker containers, which simplifies the operational eff
 
+## ECE service containers per host role(s) [ece-service-containers]
+
+Each Elastic Cloud Enterprise service runs as a dedicated Docker container. These containers are automatically deployed based on the roles assigned to each ECE host.
```
```suggestion
Each Elastic Cloud Enterprise service runs as a dedicated container. These containers are automatically deployed based on the roles assigned to each ECE host.
```

What about removing the word Docker considering we also support Podman?

shainaraskas replied:
based on this note, we might need to remove it from the section above as well ("services as docker containers")

rseldner replied:
Thanks @eedugon and @shainaraskas.  I'm all for omitting the docker references (it's mentioned 6 times on the page) and improving the entire page as a whole, but it is out of scope of my proposal of documenting the service containers.

eedugon replied:
^^ don't worry about that @rseldner , we could remove `Docker` references if needed afterwards. I just wanted to ensure we didn't add more in the PR :D

rseldner replied:
agreed. thanks!

eedugon commented on deploy-manage/deploy/cloud-enterprise/ece-architecture.md:
```diff
@@ -84,3 +84,26 @@ Services are deployed as Docker containers, which simplifies the operational eff
+| `frc-blueprints-blueprint`                          | Director      | Coordinates container startup by providing configuration data to runners based on their role and token. |
+| `frc-client-forwarders-client-forwarder`            | All roles     | Manages communication between nodes and ZooKeeper. |
+| `frc-cloud-uis-cloud-ui`                            | Controller    | Web frontend for the ECE UI, served to users in the browser. |
+| `frc-constructors-constructor`                      | Controller    | Schedules changes via ZooKeeper; assigns clusters to allocators and balances zones. |
```
```suggestion
| `frc-constructors-constructor`                      | Controller    | Schedules and coordinates deployment changes through ZooKeeper; assigns instances to allocators and balances zones. |
```

eedugon commented:
great initiative! I'm totally ok with the change and giving a brief description of each container set.

Would it be ok if I double check with someone from development team or PM about the accuracy of the `descriptions` in the table? Just to ensure all looks good.

In the meantime @rseldner , it would be great if you review and commit the suggestions made by @jakommo and myself (of course if you agree with them).

rseldner committed:
Update deploy-manage/deploy/cloud-enterprise/ece-architecture.md

thanks!

Co-authored-by: Jakob Reiter <jakommo@users.noreply.github.com>

rseldner committed:
Update deploy-manage/deploy/cloud-enterprise/ece-architecture.md

Thanks!

Co-authored-by: Jakob Reiter <jakommo@users.noreply.github.com>

rseldner committed:
Update deploy-manage/deploy/cloud-enterprise/ece-architecture.md

thanks!

Co-authored-by: Edu González de la Herrán <25320357+eedugon@users.noreply.github.com>

rseldner commented:
Thanks for the feedback so far all 🙇 


👋 @kunisen ...
### [1]
The same containers are already publicly documented here since 3.7:
- Current: https://www.elastic.co/docs/troubleshoot/deployments/cloud-enterprise/troubleshooting-container-engines#ece-troubleshooting-containers-unhealthy-restart
- 3.7: https://www.elastic.co/guide/en/cloud-enterprise/3.7/ece-troubleshooting-containers.html

And I think ECE admins should be aware of the containers that exist in their environment and that ECE relies on even if at a basic level.  It would be particularly useful to know which containers are _expected_ to run on each host type for admins to know if something might be amiss when looking at list of running containers.


### [2]
Agreed on getting cloud devs' blessing.  Thank you!

### [3] 
**A**
Regarding the roles, that page jumps from the "Service-oriented architecture" to basically discussion of the roles.  Though it does not explicitly mention the term "role" it pretty much checks all the boxes.  
- ✔️  Control Plane
  - ✔️ Director
  - ✔️   Constructor (perhaps rename)
- ✔️  Proxies
- ✔️ Allocator

Maybe there is a better place for all of this? The page I picked seemed fitting to me based on the reason above.

**B**
Wouldn't hurt to align the rest of the documentation but the terms are a bit all over the place as you mention 😅 

**C**
Agreed on the suggestion.  Grouping first by roles instead makes sense to me.

eedugon commented:
I don't think the list of container sets should be considered too low level or ECE internals when we have public docs that already mention some of them when providing commands.

We could have this info in a KB and not make it visible in the public docs, but having a good reference between the ECE host roles and the expected containers is great in my opinion, except if we plan to change that very often and we don't want users to have clear expectations in terms of what to expect when they run docker ps.

jakommo commented:
> I don't think the list of container sets should be considered too low level or ECE internals 

Full ack. For folks administering ECE environments this is very useful info and belongs in the public docs

kunisen commented:
@przemek-grzedzielski 👋  when you have time, may I trouble you to take a look on this before we merge the PR please? thanks!

rseldner committed:
Update ece-architecture.md

reordered the list alphabetically, first based on role, second by container name.

kunisen commented:
Pinged [internally (here)](https://elastic.slack.com/archives/CS5UH0E3T/p1753950386029669) for follow up.

rseldner committed:
Merge branch 'main' into rseldner-patch-2

eedugon commented on deploy-manage/deploy/cloud-enterprise/ece-architecture.md:
```diff
@@ -86,3 +86,24 @@ Services are deployed as Docker containers, which simplifies the operational eff
 
+## ECE service containers per host role(s) [ece-service-containers]
+
+Each Elastic Cloud Enterprise service runs as a dedicated Docker container. These containers are automatically deployed based on the roles assigned to each ECE host.
```
```suggestion
Each {{ece}} service runs as a dedicated container. These containers are automatically deployed based on the roles assigned to each ECE host. The following table lists the containers you’ll find on your ECE hosts:
```

eedugon commented on deploy-manage/deploy/cloud-enterprise/ece-architecture.md:
```diff
@@ -86,3 +86,24 @@ Services are deployed as Docker containers, which simplifies the operational eff
+| `frc-client-forwarders-client-forwarder`          | All roles     | Manages communication between hosts and ZooKeeper. |
+| `frc-runners-runner`                              | All roles     | Runs on every ECE host and provides a supervisor service to deploy and manage containers according to their defined roles, ensuring they are online and healthy. |
+| `frc-services-forwarders-services-forwarder`      | All roles     | Routes internal service data across the ECE platform. |
+| `frc-allocator-metricbeats-allocator-metricbeat`  | Allocator     | Collects allocator metrics via Beats. |
```
```suggestion
| `frc-allocator-metricbeats-allocator-metricbeat`  | Allocator     | Collects allocator metrics. |
```

eedugon commented on deploy-manage/deploy/cloud-enterprise/ece-architecture.md:
```diff
@@ -86,3 +86,24 @@ Services are deployed as Docker containers, which simplifies the operational eff
+| `frc-runners-runner`                              | All roles     | Runs on every ECE host and provides a supervisor service to deploy and manage containers according to their defined roles, ensuring they are online and healthy. |
+| `frc-services-forwarders-services-forwarder`      | All roles     | Routes internal service data across the ECE platform. |
+| `frc-allocator-metricbeats-allocator-metricbeat`  | Allocator     | Collects allocator metrics via Beats. |
+| `frc-allocators-allocator`                        | Allocator     | Manages container lifecycle for Elasticsearch and Kibana; reports host capacity to ZooKeeper. |
```
```suggestion
| `frc-allocators-allocator`                        | Allocator     | Manages container lifecycle for {{es}} and {{kib}} instances; reports host capacity to ZooKeeper. |
```

eedugon commented on deploy-manage/deploy/cloud-enterprise/ece-architecture.md:
```diff
@@ -86,3 +86,24 @@ Services are deployed as Docker containers, which simplifies the operational eff
+| `frc-allocator-metricbeats-allocator-metricbeat`  | Allocator     | Collects allocator metrics via Beats. |
+| `frc-allocators-allocator`                        | Allocator     | Manages container lifecycle for Elasticsearch and Kibana; reports host capacity to ZooKeeper. |
+| `frc-container-task-services-container-task-service` | Allocator  | Supports autoscaling and tracks feature usage. |
+| `frc-admin-consoles-admin-console`                | Controller    | Backend service for the ECE UI; handles API requests and coordinates with ZooKeeper, Elasticsearch, logging, and security services. |
```
```suggestion
| `frc-admin-consoles-admin-console`                | Controller    | Backend service for the ECE UI; handles API requests and coordinates with ZooKeeper, {{es}}, logging, and security services. |
```

eedugon reviewed (dismissed):
Some comments added, looks very good in general and I think it's a good addition.

rseldner committed:
Update deploy-manage/deploy/cloud-enterprise/ece-architecture.md

TY!

Co-authored-by: Edu González de la Herrán <25320357+eedugon@users.noreply.github.com>

rseldner committed:
Update deploy-manage/deploy/cloud-enterprise/ece-architecture.md

TY!

Co-authored-by: Edu González de la Herrán <25320357+eedugon@users.noreply.github.com>

rseldner committed:
Update deploy-manage/deploy/cloud-enterprise/ece-architecture.md

TY!

Co-authored-by: Edu González de la Herrán <25320357+eedugon@users.noreply.github.com>

rseldner committed:
Update deploy-manage/deploy/cloud-enterprise/ece-architecture.md

TY!

Co-authored-by: Edu González de la Herrán <25320357+eedugon@users.noreply.github.com>

rseldner committed:
Change blueprint service role from Director to Controller

corrected the role for `frc-blueprints-blueprint`

rseldner committed:
Merge branch 'main' into rseldner-patch-2

rseldner committed:
Merge branch 'main' into rseldner-patch-2

alxchalkias reviewed (approved):
One small suggestion to disambiguate "roles" from user roles/RBAC. Other than that LGTM.

kunisen committed:
Merge branch 'main' into rseldner-patch-2

rseldner committed:
Update deploy-manage/deploy/cloud-enterprise/ece-architecture.md

makes sense. thanks!

Co-authored-by: Alex Chalkias <34575586+alxchalkias@users.noreply.github.com>

przemek-grzedzielski commented on deploy-manage/deploy/cloud-enterprise/ece-architecture.md:
```diff
@@ -86,3 +86,24 @@ Services are deployed as Docker containers, which simplifies the operational eff
+|---|---|---|
+| `frc-beats-runners-beats-runner`                  | All roles     | Collects metrics from local containers for monitoring and health checks. |
+| `frc-client-forwarders-client-forwarder`          | All roles     | Manages communication between hosts and ZooKeeper. |
+| `frc-runners-runner`                              | All roles     | Runs on every ECE host and provides a supervisor service to deploy and manage containers according to their defined roles, ensuring they are online and healthy. |
```
`according to their defined roles` - this suggests the _containers_ have roles defined, whereas it's _hosts_ the roles pertain to. 
In other words, a host has a role assigned to it and that role defines what containers should run on the host. 

The runner does indeed manage these containers, makes sure they're started at the proper version, etc.  , but saying that it `ensures they are online and healthy` isn't very accurate. I'd rephrase this whole line to sth like:
```
Runs on every ECE host and provides a supervisor service to deploy and manage containers based on the host's role. Ensures the required containers are started at the proper version.
```

eedugon replied:
done

przemek-grzedzielski commented on deploy-manage/deploy/cloud-enterprise/ece-architecture.md:
```diff
@@ -86,3 +86,24 @@ Services are deployed as Docker containers, which simplifies the operational eff
+| `frc-constructors-constructor`                    | Controller    | Schedules and coordinates deployment changes through ZooKeeper; assigns instances to allocators and balances zones. |
+| `frc-blueprints-blueprint`                        | Controller    | Coordinates container startup by providing configuration data to runners based on their role and token. |
+| `frc-directors-director`                          | Director      | Manages ZooKeeper and internal certificates; maintains Stunnel and quorum. |
+| `frc-zookeeper-servers-zookeeper`                 | Director      | Consistent distributed store used to track ECE state and coordinate communication between services. |
```
`store` --> `data store` ?

przemek-grzedzielski commented on deploy-manage/deploy/cloud-enterprise/ece-architecture.md:
```diff
@@ -86,3 +86,24 @@ Services are deployed as Docker containers, which simplifies the operational eff
+| `frc-blueprints-blueprint`                        | Controller    | Coordinates container startup by providing configuration data to runners based on their role and token. |
+| `frc-directors-director`                          | Director      | Manages ZooKeeper and internal certificates; maintains Stunnel and quorum. |
+| `frc-zookeeper-servers-zookeeper`                 | Director      | Consistent distributed store used to track ECE state and coordinate communication between services. |
+| `frc-proxies-proxyv2`                             | Proxy         | Routes user traffic; uses ZooKeeper to map deployment IDs to cluster nodes. |
```
I'd say `Routes user traffic to clusters (Elasticsearch, Kibana, etc.)`. 
The part about using ZooKeeper is also a bit of an implementation detail I think.

eedugon replied:
Done, but used {{stack}} deployments instead of clusters. Feel free to comment on that if you think we should change it.

przemek-grzedzielski commented on deploy-manage/deploy/cloud-enterprise/ece-architecture.md:
```diff
@@ -86,3 +86,24 @@ Services are deployed as Docker containers, which simplifies the operational eff
+| `frc-directors-director`                          | Director      | Manages ZooKeeper and internal certificates; maintains Stunnel and quorum. |
+| `frc-zookeeper-servers-zookeeper`                 | Director      | Consistent distributed store used to track ECE state and coordinate communication between services. |
+| `frc-proxies-proxyv2`                             | Proxy         | Routes user traffic; uses ZooKeeper to map deployment IDs to cluster nodes. |
+| `frc-proxies-route-server`                        | Proxy         | Manages internal routing and port mapping for cluster access. |
```
`Manages internal routing` may not be very accurate I think? We could say `Manages routing tables used by proxyv2 to route traffic to clusters`.

eedugon replied:
Made it a bit generic:
> Manages the routing tables used by the proxy service.

Feel free to comment on that

przemek-grzedzielski commented:
This is very useful indeed, thank you for putting this together!

eedugon commented:
@przemek-grzedzielski , thanks a lot for reviewing this.

My latest question is about the order of the elements. Have we put some thought into it @rseldner or @kunisen ?

Currently the order looks a bit random, not sure if it would be better an alphabetical order, or an order based on some conceptual logic (like starting with the runner itself and then continuing with each of the roles and ending with common / auxiliary containers).

rseldner committed:
Update ECE service containers description

Clarified the description of ECE service containers and their deployment based on host roles.

rseldner commented on deploy-manage/deploy/cloud-enterprise/ece-architecture.md:
```diff
@@ -86,3 +86,24 @@ Services are deployed as Docker containers, which simplifies the operational eff
+
+Each {{ece}} service runs as a dedicated container. These containers are automatically deployed based on the roles assigned to each ECE host. The following table lists the containers you’ll find on your ECE hosts:
+
+| Container                                          | Host roles         | Description |
```
updated with `The following table lists the containers on ECE hosts, along with the host roles that include each container.`

eedugon commented:
@przemek-grzedzielski , I like your idea a lot, but, in order to be efficient, let's try to start publishing the content with the text improvements you have suggested, and in a future iteration we could just re-order the elements and perhaps add an extra column for the category, so the table feels even more informative.

I'd prefer to not bother @rseldner with extra changes considering what we have is already beneficial (sorry for having raised that Roberto ;) ).

kunisen commented:
I also had a talk with @yang-wei that we may also want to document some traffic route explanation, once we have this PR merged. (we need this PR describing the roles & components.)

<details>

The motivation is that, Yangwei brought it up that, it would be good to make it clear that ES nodes talk to each other without going thru proxy, while Kibana talking to ES nodes going thru proxy, etc. 
Having this piece of traffic information could help improve understand of our customers and ourselves well. 

<img width="1552" height="1031" alt="image" src="https://github.com/user-attachments/assets/64f5a873-767d-4421-95e8-2788244cadf1" />

</details>

We will also discuss with PMs first before we document this. 
I want to put it here for future reference.

eedugon commented on deploy-manage/deploy/cloud-enterprise/ece-architecture.md:
```diff
@@ -86,3 +86,24 @@ Services are deployed as Docker containers, which simplifies the operational eff
+
+| Container                                          | Host roles         | Description |
+|---|---|---|
+| `frc-beats-runners-beats-runner`                  | All roles     | Collects metrics from local containers for monitoring and health checks. |
```
done

eedugon commented on deploy-manage/deploy/cloud-enterprise/ece-architecture.md:
```diff
@@ -86,3 +86,24 @@ Services are deployed as Docker containers, which simplifies the operational eff
+| `frc-cloud-uis-cloud-ui`                          | Controller    | Web frontend for the ECE UI, served to users in the browser. |
+| `frc-constructors-constructor`                    | Controller    | Schedules and coordinates deployment changes through ZooKeeper; assigns instances to allocators and balances zones. |
+| `frc-blueprints-blueprint`                        | Controller    | Coordinates container startup by providing configuration data to runners based on their role and token. |
+| `frc-directors-director`                          | Director      | Manages ZooKeeper and internal certificates; maintains Stunnel and quorum. |
```
done

eedugon committed:
implementing suggestions

eedugon committed:
Merge branch 'main' into rseldner-patch-2

eedugon committed:
implementing suggestions

eedugon committed:
Merge branch 'rseldner-patch-2' of github.com:elastic/docs-content into rseldner-patch-2

eedugon committed:
small reordering

eedugon committed:
Merge branch 'main' into rseldner-patch-2

eedugon commented on deploy-manage/deploy/cloud-enterprise/ece-architecture.md:
```diff
@@ -86,3 +86,24 @@ Services are deployed as Docker containers, which simplifies the operational eff
 
+## ECE service containers per host role(s) [ece-service-containers]
+
+Each {{ece}} service runs as a dedicated container. These containers are automatically deployed based on the roles assigned to each ECE host. The following table lists the containers on ECE hosts, along with the host roles that include each container.:
```
```suggestion
Each {{ece}} service runs as a dedicated container. These containers are automatically deployed based on the roles assigned to each ECE host. The following table lists the containers on ECE hosts, along with the host roles that include each container:
```

eedugon committed:
Update deploy-manage/deploy/cloud-enterprise/ece-architecture.md

eedugon committed:
Update deploy-manage/deploy/cloud-enterprise/ece-architecture.md

eedugon committed:
Merge branch 'main' into rseldner-patch-2
